### ISSUE TITLE
'Use Dark Theme' is useless on Plasma DE

### ISSUE THREAD

**Issue by Steinhagen** [lutris/lutris#2761]
**Describe the bug**
The 'Use Dark Theme' doesn't do anything on a Plasma Desktop Environment (KDE's DE).

**Expected behavior**
Separate the application theme only for Lutris.
If that is not possible under the Plasma DE, I would expect that option to not be displayed.

**Current behavior**
It's just a toggle button that doesn't do anything.

**Steps to reproduce**
1. Use a Plasma 5 Desktop Environment
2. Open Lutris from the command line.
3. Open the 'Manage runners' window.
4. Make sure you don't have 'Atari800' runner installed.
5. Click the 'View Options' button for the application window.
6. Toggle the 'Use Dark Theme' button.

**Comment by VanackSabbadium** [comment]
It happens to me too.
I've got AwesomeVM with Compton enabled. Ticking Dark Mode doesn't do anything.

**Comment by xordspar0** [comment]
I haven't tested everything, but I think it only works for some GTK themes. Even on Gnome, it doesn't work if I'm using some themes. I can get it to work with Breeze though. What theme are you using on Plasma, @Steinhagen?

What works:
- Adwaita
- Arc
- Breeze

What doesn't work:
- Adwaita-dark
- Arc-dark
- HighContrast
- Breeze-dark

**Comment by VanackSabbadium** [comment]
> I haven't tested everything, but I think it only works for some GTK themes. Even on Gnome, it doesn't work if I'm using some themes. I can get it to work with Breeze though. What theme are you using on Plasma, @Steinhagen?
> 
> What works:
> 
> * Adwaita
> * Arc
> * Breeze
> 
> What doesn't work:
> 
> * Adwaita-dark
> * Arc-dark
> * HighContrast
> * Breeze-dark

Oh well, i think that's the problem. I have no dark theme installed. I've got a customized theme for AwesomeWM, and nothing else since i have no gnome or KDE installed.

**Comment by xordspar0** [comment]
Digging in more, I see that Lutris just sets the GTK property "gtk-application-prefer-dark-theme" when that box is checked: [lutris/gui/lutriswindow.py:452](https://github.com/lutris/lutris/blob/master/lutris/gui/lutriswindow.py#L452)

I'm not sure yet how that is implemented in GTK, but [the docs](https://developer.gnome.org/gtk3/unstable/GtkSettings.html#GtkSettings--gtk-application-prefer-dark-theme) say:

> If a GTK+ theme includes a dark variant, it will be used instead of the configured theme.

I'm not sure how a theme declares that it has a dark variant to GTK, but in my experiments all of the dark variants had the same name as the light variant with "-dark" appended to the end.

> I have no dark theme installed. I've got a customized theme for AwesomeWM

Unfortunately, there's no magic. This feature just switches to a different theme. So you could *make* a dark theme for your custum theme, I guess. :confused: 

> The 'Use Dark Theme' doesn't do anything on a Plasma Desktop Environment

When I install the breeze-gtk package on Arch, it comes with a light and dark GTK theme. I'm also able to switch between the dark and light themes.

The strange this is that when I use KDE Neon with the default packages installed, dark mode does not work, as reported. But KDE Neon comes with Breeze dark and light variants for GTK. I can see them in `/usr/share/themes/Breeze/gtk-3.0/{gtk.css,gtk-dark.css}`. Those are the same files that I have on my Arch install.

The TL;DR of this is that Lutris just sets a GTK property that it wants a dark theme. The details of how that happens are left to GTK, and I don't clearly understand how GTK works.

**Comment by strycore** [comment]
This is not caused by Lutris. Our dark theme works.
